core/system/src/page_allocator_faults.c

```diff
@@ -94,3 +119,30 @@ uint8_t page_allocator_iterate_active_pages(int (*callback)(uint32_t start_addr,
 
     return count;
 }
+
+uint8_t page_allocator_iterate_active_page_masks(int (*callback)(uint8_t mask, uint8_t index), PageAllocatorIteratorDirection direction)
```

Comment:
@niklas-arm can you please use a globally defined type for the callback function?